app/views/games/_chessboard.html.erb

```diff
@@ -1,42 +1,61 @@
 <script>
-
- $(function(){
-
-  $(".piece").draggable({
-    update: function(event, ui){
-
-
-    }
-  })
- });
+$(document).ready(function () {
+  $(function() {
+    $('.piece').draggable({
+      snap: '.square',
+      snapMode: 'outer',
+      snapTolerance: 40,
+      opacity: 0.75,
+      cursor: 'crosshair',
+      containment: 'table',
+      revert: 'invalid'
+    });
+
+    $('.square').droppable({
+      accept: '.piece',
+      drop: function(event, ui) {
+        var piece = ui.draggable
+        var destination_square = $(this);
+
+        var update_piece = {
+          id: piece.data('piece-id'),
+          x_position: destination_square.data('data-x-position'),
+          y_position: destination_square.data('data-y-position')
+        }
+
+        $.ajax({
+          type: 'PATCH',
+          url: '/pieces/' + piece.id,
```

Comment:
You will need to add an additional line here:

```javascript
beforeSend: $.rails.CSRFProtection,
```

Rails prevents cross-site request forging (CSRF) with a token.  A short description of CSRF is where a webpage tricks your browser into sending a request it did not intend to another server.  Like suppose you had www.badsite.com that had an image tag like so:

```html
<img src="https://www.facebook.com/user/delete-account/confirm" />
```

If that was the real delete-your-account URL and Facebook didn't have CSRF protection, that might cause anybody who visited that webpage to suddenly lose their FB account.

That's a simple example, and it can get more complex...

Rails solves this by generating a unique token for every page load and requiring that token to be present for any PUT / PATCH / POST / DELETE request.  That's what this line is outputting: https://github.com/team-rook/chess-app/blob/bc17f18c170d708a01e8adb35f6c9d10bf84c33b/app/views/layouts/application.html.erb#L7

The above line will cause jQuery to include that special token in your ajax request here - otherwise Rails will ignore it as a security measure.